UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.IntegrationTests/NoticeToMariners/NoticeToMarinersControllersTest.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.DependencyInjection;
+using NUnit.Framework;
+using UKHO.MaritimeSafetyInformation.Common.Models.NoticesToMariners;
+using UKHO.MaritimeSafetyInformation.Web;
+using UKHO.MaritimeSafetyInformation.Web.Controllers;
+
+namespace UKHO.MaritimeSafetyInformation.IntegrationTests.NoticesToMariners
+{
+    class NoticeToMarinersControllersTest
+    {
+        readonly IServiceProvider _services = Program.CreateHostBuilder(Array.Empty<string>()).Build().Services;
+
+        private NoticesToMarinersController _nMController;
+
+        private IHttpContextAccessor _contextAccessor;
+        private HttpContext _context;
+        private Configuration Config { get; set; }
+
+        [SetUp]
+        public void Setup()
+        {
+            Config = new Configuration();
+            _context = new DefaultHttpContext();
+            _contextAccessor = new HttpContextAccessor
+            {
+                HttpContext = _context
+            };
+            _nMController = ActivatorUtilities.CreateInstance<NoticesToMarinersController>(_services);
+        }
+
+        [Test]
+        public async Task WhenCallIndexOnLoad_ThenReturnList()
+        {
+            IActionResult result = await _nMController.Index();
+            ShowWeeklyFilesResponseModel showWeeklyFiles = (ShowWeeklyFilesResponseModel)((ViewResult)result).Model;
```

Comment:
Since in main code logic the week value was not fetching for that particular year, the showresponsefilelist count was zero. Have made necessary change in code logic and now added the assert as well